### PR TITLE
Added option to select download format.

### PR DESCRIPTION
**Pull request by rohittp0**
1.  Added these formats for users to select from when downloading images. 
     - SVG
     - PNG

2. Added check for blank college name.

Tested on chrome and firefox (latest version).

The idea inspired by @nikiljos comment on #21

### CODE REVIEW

**Comment by rohittp0** [comment]
Live at https://rohittp0.github.io/campus-logo-generator/

**Comment by nikiljos** [comment]
2 copies of the same file getting downloaded for JPEG and PNG formats

**Comment by nikiljos** [comment]
image
Couldn't we avoid an over-sized extra copy being formed here for PNG and JPEG?

**Comment by akhilmhdh** [comment]
Some changes that I recommend

1. Showing the default as SVG in the button on first render will be better to avoid confusion.
2. Avoid JPEG, it is unnecessary in both terms of size and quality, SVG and PNG would suffice most of the situation. Even SVG only would still be lot enough.
3. Add a validation in form to have at least any value.

**Comment by rohittp0** [comment]
> image
> Couldn't we avoid an over-sized extra copy being formed here for PNG and JPEG?

😂 You got me just in time. Was fixing it and when I looked here saw your comment. It's now fixed.

**Comment by nikiljos** [comment]
Why not give `Foundation` as the default value? So we would be downloading the main logo if we don't have anything specific

**Comment by rohittp0** [comment]
> Some changes that I recommend
> 
> 1. Showing the default as SVG in the button on first render will be better to avoid confusion.
> 2. Avoid JPEG, it is unnecessary in both terms of size and quality, SVG and PNG would suffice most of the situation. Even SVG only would still be lot enough.
> 3. Add a validation in form to have at least any value.


 2. I added JPEG since it would provide a background ( especially when using darkmode )
 3. Not sure what you mean, can you please explain.

**Comment by rohittp0** [comment]
> Why not give `Foundation` as the default value? So we would be downloading the main logo if we don't have anything specific

Are you talking about the download button? If so , I tried and it makes the button text go multiple lies or makes the button awkwardly long. Just putting Foundation or SVG instead of Download SVG may cause more confusion ( I think ) since there will be no indication as to what the button does.

**Comment by akhilmhdh** [comment]
1. No i meant Download(SVG)  something like this not just the word SVG.
2. Why do we need a background. The logo doesn't have a background and its designed in that way.
3. A simple form validation when textfield is empty just disable the download button.

**Comment by nikiljos** [comment]
> > Why not give `Foundation` as the default value? So we would be downloading the main logo if we don't have anything specific
> 
> Are you talking about the download button? If so , I tried and it makes the button text go multiple lies or makes the button awkwardly long. Just putting Foundation or SVG instead of Download SVG may cause more confusion ( I think ) since there will be no indication as to what the button does.

I didn't mean the download button, I said about the college name part... If no college name is entered the default coulf be foundation right?

**Comment by rohittp0** [comment]
> 1. No i meant Download(SVG)  something like this not just the word SVG.
> 2. Why do we need a background. The logo doesn't have a background and its designed in that way.
> 3. A simple form validation when textfield is empty just disable the download button.

 1. ✅
 2. ✅
 3. ✅
 
Can you check if everything is ok now?

**Comment by rohittp0** [comment]
> 1. No i meant Download(SVG)  something like this not just the word SVG.
> 2. Why do we need a background. The logo doesn't have a background and its designed in that way.
> 3. A simple form validation when textfield is empty just disable the download button.

 1. ✅
 2. ✅
 3. ✅
 
Can you check if everything is ok now?

**Comment by rohittp0** [comment]
> 1. No i meant Download(SVG)  something like this not just the word SVG.
> 2. Why do we need a background. The logo doesn't have a background and its designed in that way.
> 3. A simple form validation when textfield is empty just disable the download button.

 1. ✅
 2. ✅
 3. ✅
 
Can you check if everything is ok now?

**Comment by rohittp0** [comment]
> 1. No i meant Download(SVG)  something like this not just the word SVG.
> 2. Why do we need a background. The logo doesn't have a background and its designed in that way.
> 3. A simple form validation when textfield is empty just disable the download button.

 1. ✅
 2. ✅
 3. ✅
 
Can you check if everything is ok now? 

> > > Why not give `Foundation` as the default value? So we would be downloading the main logo if we don't have anything specific
> > 
> > 
> > Are you talking about the download button? If so , I tried and it makes the button text go multiple lies or makes the button awkwardly long. Just putting Foundation or SVG instead of Download SVG may cause more confusion ( I think ) since there will be no indication as to what the button does.
> 
> I didn't mean the download button, I said about the college name part... If no college name is entered the default coulf be foundation right?

I just tried that and it seems to be confusing. Because usually, we might expect to download an image with no text when clicking download without entering anything.

**Comment by nikiljos** [comment]
The light/dark theme button is intended to toggle the theme right? Is downloading just when we click that button necessary... ISn't it confusing to have something downloaded just when we try to toggle the theme and preview

**Comment by LunaticHacker** [comment]
add a type attribute to button  in [L35](https://github.com/tinkerhub/campus-logo-generator/blob/77458410205909f26c0ab0c18dfcd27f4e89f05a/index.html#L35) to prevent it from submitting on  click

**Comment by rohittp0** [comment]
> The light/dark theme button is intended to toggle the theme right? Is downloading just when we click that button necessary... ISn't it confusing to have something downloaded just when we try to toggle the theme and preview

Sorry this is a bug.

**Comment by rohittp0** [comment]
> add a type attribute to button in [L35](https://github.com/tinkerhub/campus-logo-generator/blob/77458410205909f26c0ab0c18dfcd27f4e89f05a/index.html#L35) to prevent it from submitting on click

Done

**Comment by nikiljos** [comment]
image
**BUG** 
_Steps to reproduce_

- Type a name(say test)
- change to PNG and click download
- Now change the name to something else(say college)
- Click download and you will get the new name above the old one

File: scripts/main.js [hunk 1]
```diff
@@ -35,19 +57,40 @@ function triggerDownload(imgURI) {
...
+        return triggerDownload(blobURL);
+ 
+    image.addEventListener("load", function gotImage() {
+
+        window.URL.revokeObjectURL(blobURL);
+        image.removeEventListener("load",gotImage);
+
+        var ctx = canvas.getContext("2d");
+
+        ctx.drawImage(image,0,0);
```
Comment (LunaticHacker) [inline scripts/main.js:87]: we should clear the canvas before drawing

Comment (rohittp0) [reply scripts/main.js:87]: Done

**Comment by rohittp0** [comment]
> image
> **BUG**
> _Steps to reproduce_
> 
> * Type a name(say test)
> * change to PNG and click download
> * Now change the name to something else(say college)
> * Click download and you will get the new name above the old one

Fixed this

**Comment by nikiljos** [comment]
Hey why is it not yet merged 🤔 It's about a month

**Pull request merged**